.github/workflows/ibis-backends.yml

```diff
@@ -760,6 +772,10 @@ jobs:
       - name: run tests
         run: just ci-check -m ${{ matrix.backend.name }} --numprocesses auto --dist=loadgroup
 
+      - name: check that no untracked files were produced
+        shell: bash
+        run: git checkout poetry.lock pyproject.toml && ! git status --porcelain | tee /dev/stderr | grep .
+
```

Comment:
this seems like the conclusion to a frustrating story

Reply:
Yeah ... ideally we don't have to modify the pyproject.toml with any `poetry add` commands, but we're bound to pyarrow 10 in the lock file because of `$REASONS`.

Reply:
I add the latest pyarrow so that we can test against it. Once snowflake-connector-python moves to nanoarrow at least the `git checkout` bits can go away.